cdap-unit-test/src/main/java/io/cdap/cdap/test/UnitTestManager.java

```diff
@@ -203,6 +203,8 @@ public ApplicationManager deployApplication(NamespaceId namespace, Class<? exten
   @Override
   public ApplicationManager deployApplication(ApplicationId appId, AppRequest appRequest) throws Exception {
     appFabricClient.deployApplication(appId, appRequest);
+    ApplicationDetail appDetail = appFabricClient.getInfo(appId);
+    appId = new ApplicationId(appId.getNamespace(), appId.getApplication(), appDetail.getAppVersion());
```

Comment:
nit: It's generally better not to change parameter values, but introduce a new variable. It's too easy to overlook such change